lib/reporters/console.js

```diff
@@ -209,8 +209,19 @@ internals.Reporter.prototype.end = function (notebook) {
             if (test.err.actual !== undefined &&
                 test.err.expected !== undefined) {
 
-                const actual = internals.stringify(test.err.actual);
-                const expected = internals.stringify(test.err.expected);
+                let actual = internals.stringify(test.err.actual);
+                let expected = internals.stringify(test.err.expected);
+
+                if (this.settings['max-diff-length'] > 0) {
+                    const maxLen = this.settings['max-diff-length'];
+                    if (actual.length > maxLen) {
+                        actual = actual.substr(0, maxLen) + '[truncated]';
+                    }
+
+                    if (expected.length > maxLen) {
+                        expected = expected.substr(0, maxLen) + '[truncated]';
+                    }
```

Comment:
Shouldn't we test for `(actual.length + expected.length) > maxLen`? In the end with the current code you can print `2 * 20 * 2013` to the console. Isn't it too much or did you already took that into account?

Reply:
>  Isn't it too much or did you already took that into account?

The limit is mostly about not overwhelming the diff algorithm (as it gets very slow for larger strings). Since both strings are diff’d separately, the limit can be applied separately.

Since this is not a hard limit like a buffer size, I don’t think it makes any sense to add any complexity to take the suffix into account.